07-complex-data-type/src/ComplexDataTypeSample.java

```diff
@@ -0,0 +1,281 @@
+import com.terracottatech.store.Cell;
+import com.terracottatech.store.CellSet;
+import com.terracottatech.store.ComplexData;
+import com.terracottatech.store.ComplexRecord;
+import com.terracottatech.store.Dataset;
+import com.terracottatech.store.DatasetWriterReader;
+import com.terracottatech.store.Record;
+import com.terracottatech.store.StoreException;
+import com.terracottatech.store.StoreList;
+import com.terracottatech.store.StoreMap;
+import com.terracottatech.store.Type;
+import com.terracottatech.store.configuration.DatasetConfiguration;
+import com.terracottatech.store.definition.ComplexDataParseException;
+import com.terracottatech.store.definition.ComplexDataParser;
+import com.terracottatech.store.manager.DatasetManager;
+
+import java.net.URI;
+import java.util.Optional;
+
+public class ComplexDataTypeSample {
+
+    private static final String TERRACOTTA_URI_ENV = "TERRACOTTA_SERVER_URL";
+    private static final String DEFAULT_TSA_PORT = "9410";
+    private static final String DEFAULT_SERVER_URI_STR = "terracotta://localhost:" + DEFAULT_TSA_PORT;
+    private static final String DATASET_NAME = "mySampleDataset";
+    private static final String SERVER_RESOURCE = "main";
+    private static final String SERVER_URI_STR = System.getenv(TERRACOTTA_URI_ENV) == null ? DEFAULT_SERVER_URI_STR : System.getenv(TERRACOTTA_URI_ENV);
+
+    public static void main(String[] args) throws ComplexDataParseException, StoreException {
+
+        try (DatasetManager datasetManager = DatasetManager.clustered(URI.create(SERVER_URI_STR)).build()) {
+            // clean-up (delete) sample dataset from any previous run of this sample program
+            datasetManager.destroyDataset(DATASET_NAME);
+
+            // create and use a dataset
+            DatasetConfiguration offheapResource = datasetManager.datasetConfiguration().offheap(SERVER_RESOURCE).build();
+            datasetManager.newDataset(DATASET_NAME, Type.LONG, offheapResource);
+
+            try (Dataset<Long> rawDataset = datasetManager.getDataset(DATASET_NAME, Type.LONG)) {
+
+                DatasetWriterReader<Long> myDataset = rawDataset.writerReader();
+                myDataset.add(123L, originalCellSet());
+
+                System.out.println("int value retrieved from Dataset is : "
```

Comment:
The integration with the dataset is fine here, but it needs to be extended to the rest of the sample code. Of particular importance is use of the functional types (maps, filters) with the streams returned from the dataset.

Reply:
We're still missing any examples showing the integration of the CDT types with streams of records. You can take a look at the three test methods here: https://github.softwareag.com/Terracotta/terracotta-enterprise/main/store/embedded/src/test/java/com/terracottatech/store/MapListIT.java for some examples of the sort of thing I'm talking about.